### DISCUSSION TITLE
Duda/ayuda TP 6 ejercicio esta_ordenado?

### DISCUSSION THREAD

**Discussion by NicoHermosilla** [INGCOM-UNRN-P1/cursada-2023#185] [Preguntas y respuestas] [answered]
Buenas estoy teniendo problemas para plantear la situacion en la que el primer numero de mi archivo dice cuantos numeros contiene este.
Si bien el codigo que hice funciona y dice si esta ordenado o no. No estaria pudiendo implentar un caso en el que no se incluya el primer numero de mi archivo (adjunto imagen para aclarar por si no se entiende):
![image](https://github.com/INGCOM-UNRN-P1/cursada-2023/assets/105023590/d7ab9ba6-4876-4b5e-85c1-1fccced42596)
https://github.com/INGCOM-UNRN-P1/tp6-2023-NicoHermosilla

**Comment by vlzJuan** [comment, marked as answer]
Respecto a lo que decis, ¿El numero de la primera linea tambien se considera para ver si está ordenado o no? ¿O lo estas guardando 'aparte'?

Estoy entendiendo tu consulta como el primer caso que propuse (testeas todos los numeros). Mi solucion fue lo que puse de segunda pregunta (guardarlo aparte)

Una vez tengas el puntero al archivo, asumo que lo vas leyendo linea por linea. En mi caso, para hacerlo use la funcion 
```C
fgets(linea, MAX_CHAR, archivo);
```
para copiar cada linea del archivo en mi 'linea' auxiliar (declarada como ```char linea[MAX_CHAR]```, con tope de caracteres ```MAX_CHAR```, del puntero ```archivo```, que inicialmente apunta a la primera linea del archivo. Sea mi archivo de ejemplo:
```txt
5
0
1
2
3
4
```
inicialmente, ```archivo``` te lleva a la linea del tamanio (5).
tras un uso de ```fgets```, ```archivo``` te lleva a la linea del primer elemento (0)
tras otro uso, te lleva a la linea de (1), y asi. 

Asumo que, estes usando esa funcion o no, estas trabajando con alguna similar que corre el puntero a 'la linea siguiente' cada vez que lo usas.
Con usar tu 'recibir linea' una vez, ya estarias viendo el primer elemento que queres empezar a analizar. Podes usar tu codigo que verifica si esta ordenado o no desde ahi.

**Reply by NicoHermosilla** [reply]
No habia pensado en guardar la primer linea, capaz sea por eso que no lograba encontrar la forma de verificar sin tener en cuenta eso.

**Comment by martinvilu** [comment]
Hace un `fscanf` para leer la primera línea con el número, y después mete el `fgets` para leer todo (o casi todo) el resto.

**Reply by NicoHermosilla** [reply]
Creo que logre resolverlo almacenando mi primer linea en una variable la cual ira disminuyendo a medida que se recorra el bucle comparador, ademas agrege un caso en el cual si mi archivo no contenie la linea que dice cuantos numeros tiene el archivo, que este diga directamente que esta no ordenado.
https://github.com/INGCOM-UNRN-P1/tp6-2023-NicoHermosilla
es valido esto?

**Reply by martinvilu** [reply]
> ademas agrege un caso en el cual si mi archivo no contenie la linea que dice cuantos numeros tiene el archivo

¿Como sabes que no tiene esta linea?

**Reply by NicoHermosilla** [reply]
Se supone que si el  primer fscanf no es "existoso" quiere decir,  o que el archivo no contiene la linea que dice cuantos numeros tiene este mismo o que la supuesta cantidad de numeros que dice tener no corresponde a los que hay en el archivo.

**Reply by martinvilu** [reply]
Pero como no podés distinguir los números entre sí, si el primer número es la cantidad del archivo, solo lo podés verificar cuando terminas de procesar la cantidad indicada por ese valor, si llegaste a `EOF`, coincidió, pero si te queda archivo para procesar, entonces el número no era correcto.

**Reply by martinvilu** [reply]
#### Ejemplo correcto
```
4
1
2
3
4
```
#### Caso incorrecto 1, se queda 'corto'

```
10
1
2
3
4
5
```
#### Caso incorrecto 2, se queda 'largo'
```
4
1
2
3
4
5
6
```

Esto aparte de que aparezcan letras en el medio.

**Reply by NicoHermosilla** [reply]
claro comprendo, pero yo entendi que si en mi archivo por ejemplo dice que tiene 5 numeros, se debia solo verficar con los supuestos 5 numeros que contenia este. Es decir que si mi archivo tiene mas de la cantidad especificada deberia dar algun error?, o si tiene menos tambien?

**Reply by martinvilu** [reply]
Sí.
